# Written by Lefteris kotsonis.
# Modeling Realsense D435 Sensor Noise for Improved 3D Reconstruction and Tracking.
sensor_processor/type: structured_light
sensor_processor/cutoff_min_depth: 0.11
sensor_processor/cutoff_max_depth: 5.0
sensor_processor/ignore_points_above: 0.1
sensor_processor/ignore_points_below: -2.0
#sensor_processor/normal_factor_a: 0.000611
#sensor_processor/normal_factor_b:  0.003587
#sensor_processor/normal_factor_c: 0.3515
#sensor_processor/normal_factor_d: 0
#sensor_processor/normal_factor_e: 1
#sensor_processor/lateral_factor: 0.01576

point_cloud_topic: "/pico_flexx/points"
map_frame_id: "map"
robot_base_frame_id: "base_link"
robot_pose_with_covariance_topic: "/mavros/vision_pose/pose_cov"
robot_pose_cache_size: 500
track_point_frame_id: "deploy_point"
track_point_x: 0.0
track_point_y: 0.0
track_point_z: 0.0

vgf_leaf_horizontal: 0.05
vgf_leaf_vertical: 0.03
#min_update_rate: 2.0
num_callback_threads: 3 
#postprocessor_num_threads: 2
length_in_x: 6.0
length_in_y: 6.0
position_x: 0.0
position_y: 0.0
resolution: 0.05
#min_variance: 0.0001
#max_variance: 0.05
#mahalanobis_distance_threshold: 2.5
#multi_height_noise: 0.001
#surface_normal_positive_axis: z
#fused_map_publishing_rate: 0.0 
enable_continuous_cleanup: false
enable_visibility_cleanup: false
visibility_cleanup_rate: 3.0
#scanning_duration: 1.0
